pkg/controller/certificates/signer/signer.go

```diff
@@ -152,7 +144,7 @@ func requestValidForSignerName(req *x509.CertificateRequest, usages []capi.KeyUs
 	case capi.KubeletServingSignerName:
 		return capihelper.IsKubeletServingCSR(req, usages)
 	case capi.KubeAPIServerClientKubeletSignerName:
-		return capihelper.IsKubeletServingCSR(req, usages)
+		return capihelper.IsKubeletClientCSR(req, usages)
```

Comment:
@liggitt should we also be checking if the username on the CSR matches the commonName of the request here? We currently check it in the approval plugin (but not in defaulting).

Reply:
no, the signerName is based only on the csr/usages, not on the submitter